### PR TITLE
Update footer and navbar to match PennyLane website

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**

The PennyLane website is updating the styling of its footer and navbar. Since the [PennyLane Sphinx Theme](https://github.com/PennyLaneAI/pennylane-sphinx-theme) inherits the structure and style of the Xanadu Sphinx Theme, the updated layout, configuration, etc. are also reflected here.

**Description of the Change:**

Updated the footer and navbar configuration and styles to match the new PennyLane website. The changes to the theme configuration are documented in the diff of the `README.rst`.

<details><summary>Navbar</summary>
<p>

Navbar

</p>
</details> 

<details><summary>Footer</summary>
<p>

Footer

</p>
</details> 

<details><summary>Mobile Navbar</summary>
<p>

Mobile Navbar Expanded

Mobile Navbar ToC Expanded

</p>
</details> 

<details><summary>Mobile Footer</summary>
<p>

Mobile Footer

</p>
</details> 

**Benefits:**

* The navbar and footer on the PennyLane Docs pages will be consistent with the main PennyLane website.

**Possible Drawbacks:**

* Xanadu OSS projects which do not fall under the PennyLane umbrella will adopt the appearance of the footer and navbar on the PennyLane website if they update their dependency on the Xanadu Sphinx Theme.
* Some accessibility features were compromised in the adoption of the new footer and navbar styles. These will be reinstated (and improved!) in a future patch release.

**Related GitHub Issues:**

None.

---

**Preview:** https://xanadu-sphinx-theme--40.org.readthedocs.build/en/40/

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,17 @@
-## Release 0.4.0 (development release)
+## Release 0.4.0 (current release)
+
+### Improvements
+
+* Updated the navbar and footer configuration and styles to match the PennyLane website.
+  [#33](https://github.com/XanaduAI/xanadu-sphinx-theme/pull/33)
```
Comment (Mandrenkov) [inline .github/CHANGELOG.md:6]:
```suggestion
  [#40](https://github.com/XanaduAI/xanadu-sphinx-theme/pull/40)
```

**Review by Lorraine-at-Xanadu — CHANGES_REQUESTED** [review]
Hey @Mandrenkov! You've done an incredible job matching the navbar in Sphinx. 

1. Learn > Glossary: Link is broken and should go to https://pennylane.ai/qml/glossary
2. When I hover on 'Learn' (for example) and then hover on 'Documentation' there's a delay when the 'Learn' menu closes causing overlap. 

image

**Comment by Mandrenkov** [comment]
Thanks for the review, @Lorraine-at-Xanadu!

> 1. Learn > Glossary: Link is broken and should go to https://pennylane.ai/qml/glossary

I assume this is meant for the PennyLane Sphinx Theme PR: https://github.com/PennyLaneAI/pennylane-sphinx-theme/pull/29.

> 2. When I hover on 'Learn' (for example) and then hover on 'Documentation' there's a delay when the 'Learn' menu closes causing overlap.

I've updated the navbar so that the dropdown menus instantly close when a user stops hovering over it.

**Comment by Mandrenkov** [comment]
Thanks for the review, @LucasSilbernagel!

> The accessibility issues are really painful

I agree the accessibility of the new navbar and footer leave much to be desired. We'll tackle it soon!

**Comment by LucasSilbernagel** [comment]
> I've updated the navbar so that the dropdown menus instantly close when a user stops hovering over it.

@Mandrenkov @Lorraine-at-Xanadu this is really annoying for users with accessibility requirements and defeats the purpose of [the accessible example we followed](https://www.w3.org/WAI/tutorials/menus/flyout/#flyoutnavkbbtn) ☹️

**Review by josh146 — APPROVED** [review]
I didn't review the code, but reviewed the PR preview. Looks and works great @Mandrenkov, super impressed how much you got it to match!

**Comment by Mandrenkov** [comment]
@LucasSilbernagel, after a discussion with @Anakhom and @AndrewGardhouse offline, we decided to reinstate the one-second delay but ensure that only one dropdown from the navbar is displayed at any given time.

**Review by ashishks0522 — APPROVED** [review]

**Pull request merged**